imx219_modes.h

```diff
@@ -784,6 +784,95 @@ struct sensor_regs imx219_stop[] = {
 	{ 0x0100, 0x00 }, // disable streaming
 };
 
+static int imx219_set_crop(const struct sensor_def *const sensor, struct mode_def *const sensor_mode,
+			   const struct raspiraw_crop *const cfg)
+{
+	// All members of cfg default to -1 if the user did not set them.
+	// Check for validity.
+	if ((cfg->height == 0) || (cfg->width == 0))
+	{
+		// 0 is not a valid value for either.
+		return -1;
+	}
+	if ((cfg->height > 2464) || (cfg->width > 3820))
```

Comment:
3280 and 2464 are used numerous times, so use ```#define```s for them.

Although shouldn't the ranges also vary based on mode? If binning/skipping is enabled, then those aren't the correct limits, or rather the numbers specified will be expected to be in binned/skipped terms, which differ from the register settings.

Also shift the computing of x_start/y_start should left/top be -1 to before this, and the two checks of 
```width > 3820``` and ```left + width > 3820``` condense into one.

Reply:
Let me think about the math for those cases.

Reply:
If ```left``` is computed before the calculation, then it is simply ```left + width > 3280```.
left can not be negative, therefore if that condition is true then width will not be > 3280 either.

Reply:
Added `#define`s.

`left` might not be set, in which case it is `-1`.  In that case I can't condense it.

I am thinking of the raspberry pi as an educational tool first and foremost.  Would you agree?  I'm new to this.  Working on this code certainly helped my understanding of imaging sensors. Anyway, that's why I am erring on the side of easy to read code and don't try to optimize it (let the compiler worry about that).

You are right, binning is not taken into account.  I added TODOs for now.  I'll follow up with another pull request once I get around to working on that.

Reply:
> left might not be set, in which case it is -1. In that case I can't condense it.

Which is why I said that the calculation for x_start/y_start needed to be done first.